### PR TITLE
Rework block fetcher

### LINKED ISSUE #20: Option to start block fetcher by default
Add option in config to start block fetcher when ODC starts, so it isn't necessary to start it manually after service restart

### LINKED ISSUE #21: Log/Save latest processed block and slot
Save info about latest fetched and processed block and slot so on service restart it won't have to process already saved datums

### PR DESCRIPTION
**Pull request by t4ccer**
- Add possibility to change filter in control API
- Add API endpoint to lookup currently processed block (Close #21)
- Add config option to start block fetcher from lastly processed block (Close #21)
- Add config option to auto-start block fetcher (Close #20)
- Change config schema for block fetcher to make sense with these changes (described in README)

### CODE REVIEW

**Comment by t4ccer** [comment]
Tested on Seabug, works as desired

File: src/Api.hs [hunk 1]
```diff
@@ -34,6 +35,11 @@ data DatumApi route = DatumApi
             -- TODO: change to JSON request
             :> ReqBody '[JSON] GetDatumsByHashesRequest
             :> Get '[JSON] GetDatumsByHashesResponse
+    , getLastBlock ::
+        route
+            :- "block"
+            :> Summary "Get currenctly processed block"
```
Comment (ngua) [inline src/Api.hs:41]: There's a typo: "currenctly"

File: src/Block/Fetch.hs [hunk 2]
```diff
@@ -119,26 +119,29 @@ receiveLoop ::
...
     case _result <$> msg of
         Nothing -> do
             logErrorNS "receiveLoop" "Error decoding FindIntersect response"
         Just (IntersectionNotFound _) -> do
-            logErrorNS "receiveLoop" "Find intersection error: Intersection not found"
+            logErrorNS "receiveLoop" "Find intersection error: Intersection not found. Consider restarting block fetcher with different block info"
         Just (IntersectionFound _ _) -> do
             logInfoNS "receiveLoop" "Find intersection: intersection found, starting RequestNext loop"
-            Async.withAsync (receiveBlocksLoop conn) $ \receiveBlocksWorker -> do
+            Async.withAsync (receiveBlocksLoop conn datumFilter) $ \receiveBlocksWorker -> do
```
Comment (ngua) [inline src/Block/Fetch.hs:137]: `datumFilter` is unused here

File: src/Block/Fetch.hs [hunk 3]
```diff
@@ -119,26 +119,29 @@ receiveLoop ::
...
                 Async.link receiveBlocksWorker
                 requestRemainingBlocks conn
                 Async.wait receiveBlocksWorker
 
+-- b3faa2bc1465aa0d8500655981108904e5a7a9339be5fc1e6a9754088a4456c1
```
Comment (ngua) [inline src/Block/Fetch.hs:142]: What are these comments for?

**Review by ngua — APPROVED** [review]

File: .github/local-formatting.sh [hunk 4]
```diff
@@ -1,4 +1,4 @@
 # Extensions necessary to tell fourmolu about 
 EXTENSIONS="-o -XTypeApplications -o -XTemplateHaskell -o -XImportQualifiedPost -o -XPatternSynonyms -o -fplugin=RecordDotPreprocessor"
 SOURCES=$(git ls-tree -r HEAD --full-tree --name-only | grep -E '.*\.hs')
-~/.local/bin/fourmolu --mode inplace --check-idempotence $EXTENSIONS $SOURCES
+fourmolu --mode inplace --check-idempotence $EXTENSIONS $SOURCES
```
[the remark this answers is not in the document]
Comment (t4ccer) [reply .github/local-formatting.sh:4]: Yeah, but it's a file that formats files inplace, is there a way to do it with flake?

File: src/Api.hs (hunk 1 above)
Comment (t4ccer) [reply src/Api.hs:41]: That summary is actually wrong, `Get latest processed block` is correct one, datums from returned block are guaranteed to be already in database

File: src/Block/Fetch.hs (hunk 3 above)
Comment (t4ccer) [reply src/Block/Fetch.hs:142]: Some debugging leftovers

File: src/Block/Fetch.hs (hunk 2 above)
Comment (t4ccer) [reply src/Block/Fetch.hs:137]: It is, it's passed to `receiveBlocksLoop` then to `saveDatumsFromAlonzoBlock` and used to filter datums

**Comment by t4ccer** [comment]
No critical changes requested so I'll merge it to unblock further work

**Pull request merged**